gtda/externals/python/ripser_interface.py

```diff
@@ -206,31 +213,45 @@ def ripser(X, maxdim=1, thresh=np.inf, coeff=2, metric="euclidean",
         # births
         dm = sparse.coo_matrix(dm)
 
-    if sparse.issparse(dm):
-        if sparse.isspmatrix_coo(dm):
-            # If the matrix is already COO, we need to order the row and column
-            # indices lexicographically to avoid errors. See
-            # https://github.com/scikit-tda/ripser.py/issues/103
-            row, col, data = dm.row, dm.col, dm.data
-            lex_sort_idx = np.lexsort((col, row))
-            row, col, data = \
-                row[lex_sort_idx], col[lex_sort_idx], data[lex_sort_idx]
+    if sparse.issparse(dm) or collapse_edges:
+        if collapse_edges:
+            if not sparse.issparse(dm):
+                row, col, data = \
+                    gtda_collapser.flag_complex_collapse_edges_dense(dm)
+            else:
+                coo = dm.tocoo()
+                row, col, data = \
+                    gtda_collapser.flag_complex_collapse_edges_coo(coo.row,
+                                                                   coo.col,
+                                                                   coo.data)
         else:
-            # Lexicographic sorting is performed by scipy upon conversion
-            coo = dm.tocoo()
-            row, col, data = coo.row, coo.col, coo.data
+            if sparse.isspmatrix_coo(dm):
+                # If the matrix is already COO, we need to order the row and
+                # column indices lexicographically to avoid errors. See
+                # https://github.com/scikit-tda/ripser.py/issues/103
+                row, col, data = _lexsort_coo_data(dm.row, dm.col, dm.data)
+            else:
+                coo = dm.tocoo()
+                row, col, data = coo.row, coo.col, coo.data
```

Comment:
I expect that these are automatically lexsorted by scipy when running `tocoo`.

Reply:
True